### PR TITLE
[cmake] Don't repeat optional dependencies

### PR DESCRIPTION
**Pull request by fetzerch**
Commit fe50635fb7a09d1576890a5d548429d55af00221 (https://github.com/xbmc/xbmc/pull/10188) changed the behavior for optional dependencies. ON makes them a required dependency, AUTO an optional, OFF disables it. Therefore most of the defaults were incorrectly making most dependencies default.

This change removes explicit options for optional dependencies from CMakeLists.txt because they are implicitly added as AUTO in core_optional_dep and core_optional_dyload_dep.

@wsnipex: I guess the default should be AUTO on dev machines, right? And you just force it to ON when building packages.

This fixes the build on windows, where the avahi dependency doesn't exist.
Jenkins Builds:
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1516/ :x: 
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1517/ :x:
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1518/ :x: (unrelated issues)
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1519/ :x:
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1521/ :x:
- http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1523/ :x:

### CODE REVIEW

**Comment by Paxxi** [comment]
Seems to work fine locally for me. Can't say if it breaks anything on other platforms though

**Comment by fetzerch** [comment]
@FernetMenta: I think 767a74c got forgotten in the "drop deinterlace" change. Have put it here, because I want to get builds with cmake all green again.

**Comment by FernetMenta** [comment]
@fetzerch I am not a fan of just build-fixing things. The change for IMX is not complete. IMO there is no benefit in pretending that IMX has adjusted to the changes while it actually misses some love. The changes for AML and OpenMax seem ok. Nobody cares about OpenMax anyway.

**Comment by fetzerch** [comment]
@FernetMenta: Sry I wasn't aware of the state of IMX. Dropped the change for it.

File: project/cmake/CMakeLists.txt [hunk 1]
```diff
@@ -31,12 +24,9 @@ if(UNIX)
   endif()
   option(FFMPEG_PATH        "Path to external ffmpeg?" "")
   option(ENABLE_INTERNAL_CROSSGUID "Enable internal crossguid?" ON)
-  option(ENABLE_LIBUSB      "Enable libusb support?" ON)
-  option(ENABLE_OPENGL      "Enable OpenGL?" ON)
-  option(ENABLE_OPENGLES    "Enable OpenGLES?" ON)
   option(ENABLE_OPENSSL     "Enable OpenSSL?" ON)
   option(ENABLE_SDL         "Enable SDL?" OFF)
-  option(ENABLE_X11         "Enable X11 support?" ON)
+  option(ENABLE_X11         "Enable X11 support?" OFF)
```
Comment (wsnipex) [inline project/cmake/CMakeLists.txt]: keep this ON please.

File: project/cmake/CMakeLists.txt [hunk 2]
```diff
@@ -194,20 +184,21 @@ if(NOT WIN32)
     else()
       core_require_dep(Sdl)
     endif()
+    core_optional_dep(EGL)
```
Comment (wsnipex) [inline project/cmake/CMakeLists.txt]: AFAIK EGL isn't optional anymore, right @FernetMenta @fritsch

Comment (FernetMenta) [inline project/cmake/CMakeLists.txt]: correct

**Comment by mapfau** [comment]
I really don't know what you guys have changed, but there are now some issues regarding [dorid] cmake builds:

1.) cmake claims not to find ALSA / AVAHI (I changed this by swithcing to OFF in CMakeLists.txt)
2.) There are compile issues regarding AML (wich fit into the topic here):

```
[ 96%] Building CXX object build/cores/VideoPlayer/videorenderers/hwdec/CMakeFiles/videorenderers_hwdec.dir/RendererAML.cpp.o
In file included from /home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp:21:0:
/home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.h:44:25: error: 'EDEINTERLACEMODE' has not been declared
   virtual bool Supports(EDEINTERLACEMODE mode);
                         ^
/home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp:84:29: error: 'bool CRendererAML::Supports' is not a static member of 'class CRendererAML'
 bool CRendererAML::Supports(EDEINTERLACEMODE mode)
                             ^
/home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp:84:29: error: 'EDEINTERLACEMODE' was not declared in this scope
/home/pfau/dev/xbmc-android/xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/RendererAML.cpp:85:1: error: expected ',' or ';' before '{' token
 {
 ^  
```
AFAIR AML was compiled last time successfully, but questin remains, if AML should be build for [droid] because mediacodec is used by default, isn't it?

**Comment by fetzerch** [comment]
Builds are green on all platforms again: http://jenkins.kodi.tv/view/Helpers/job/BuildMulti-All/1528/
@wsnipex: good to go?

**Pull request merged**